ramls/oai-pmh.raml

```diff
@@ -0,0 +1,131 @@
+      This module supports the OAI-PMH as a means of exposing FOLIO metadata.
+
+types:
+  oai-pmh: !include schemas/OAI-PMH.xsd
+
+traits:
+  partionable: !include traits/partitionable.raml
+  selective: !include traits/selective.raml
+  metadataPrefix: !include traits/metadataPrefix.raml
+
+resourceTypes:
+  list: !include rtypes/list.raml
+
+/oai:
+  /records:
+    displayName: Records
+    type:
+      list:
+        exampleCollection: !include examples/records.sample
+    get:
+      is: [ partionable, selective, metadataPrefix ]
+      responses:
+        404:
+          description: "Item with a given ID and selective harvesting params (if any) not found"
+          body:
+            application/xml:
+              schema: oai-pmh
+              example: !include examples/no_records_error.sample
+
+    /{id}:
+      uriParameters:
+        id:
+          description: The unique identifier of the item in the repository
+          type: string
+      get:
+        description: Retrieve record with given Id
+        is: [ metadataPrefix ]
+        responses:
+          200:
+            description: "Returns item with a given ID"
+            body:
+              application/xml:
+                example: !include examples/record.sample
+                schema: oai-pmh
+          404:
+            description: "Item with a given ID and selective harvesting params (if any) not found"
+            body:
+              application/xml:
+                schema: oai-pmh
+                example: !include examples/no_record_error.sample
+          500:
+            description: "Internal server error, e.g. due to misconfiguration"
+            body:
+              text/plain:
+                example: "Internal server error, contact administrator"
+
+  /identifiers:
+    displayName: Identifiers
+    type:
+      list:
+        exampleCollection: !include examples/identifiers.sample
+    get:
+      is: [ partionable, selective, metadataPrefix ]
+      responses:
+        404:
+          description: "Item with a given ID and selective harvesting params (if any) not found"
+          body:
+            application/xml:
+              schema: oai-pmh
+              example: !include examples/no_records_error.sample
+
+  /metadata_formats:
+    displayName: MetadataFormats
+    type:
+      list:
+        exampleCollection: !include examples/metadata_formats.sample
+    get:
+      queryParameters:
+        identifier:
+          description: 'unique identifier of the item in the repository from which the record must be disseminated'
+          type: string
+          required: false
+      responses:
+        404:
+          description: "Item with a given ID and selective harvesting params (if any) not found"
+          body:
+            application/xml:
+              schema: oai-pmh
+              example: !include examples/no_record_error.sample
+        422:
```

Comment:
Add a 500 just in case

Reply:
/metadata_formats resource is of type 'list' which defines status code 500 among other things